File: app/src/main/java/com/github/vipulasri/timeline/TimeLineAdapter.java
package com.github.vipulasri.timeline;

import android.content.Context;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.github.vipulasri.timeline.model.OrderStatus;
import com.github.vipulasri.timeline.model.Orientation;
import com.github.vipulasri.timeline.model.TimeLineModel;
import com.github.vipulasri.timeline.utils.DateTimeUtils;
import com.github.vipulasri.timeline.utils.VectorDrawableUtils;
import com.github.vipulasri.timelineview.TimelineView;
import com.github.vipulasri.timelineview.Utils;

import java.util.List;

/**
 * Created by HP-HP on 05-12-2015.
 */
public class TimeLineAdapter extends RecyclerView.Adapter<TimeLineViewHolder> {

    private List<TimeLineModel> mFeedList;
    private Context mContext;
    private Orientation mOrientation;
    private boolean mWithLinePadding;
    private LayoutInflater mLayoutInflater;

    public TimeLineAdapter(List<TimeLineModel> feedList, Orientation orientation, boolean withLinePadding) {
        mFeedList = feedList;
        mOrientation = orientation;
        mWithLinePadding = withLinePadding;
    }

    @Override
    public int getItemViewType(int position) {
        return TimelineView.getTimeLineViewType(position,getItemCount());
    }

    @Override
    public TimeLineViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        mContext = parent.getContext();
        mLayoutInflater = LayoutInflater.from(mContext);
        View view;

        if(mOrientation == Orientation.HORIZONTAL) {
            view = mLayoutInflater.inflate(mWithLinePadding ? R.layout.item_timeline_horizontal_line_padding : R.layout.item_timeline_horizontal, parent, false);
        } else {
            view = mLayoutInflater.inflate(mWithLinePadding ? R.layout.item_timeline_line_padding : R.layout.item_timeline, parent, false);
        }

        return new TimeLineViewHolder(view, viewType);
    }

    @Override
    public void onBindViewHolder(TimeLineViewHolder holder, int position) {

        TimeLineModel timeLineModel = mFeedList.get(position);

        if(timeLineModel.getStatus() == OrderStatus.INACTIVE) {
            holder.mTimelineView.setMarker(VectorDrawableUtils.getDrawable(mContext, R.drawable.ic_marker_inactive, android.R.color.darker_gray));
        } else if(timeLineModel.getStatus() == OrderStatus.ACTIVE) {
            holder.mTimelineView.setMarker(VectorDrawableUtils.getDrawable(mContext, R.drawable.ic_marker_active, R.color.colorPrimary));
        } else {
            holder.mTimelineView.setMarker(ContextCompat.getDrawable(mContext, R.drawable.ic_marker), ContextCompat.getColor(mContext, R.color.colorPrimary));
        }

        if(!timeLineModel.getDate().isEmpty()) {
            holder.mDate.setVisibility(View.VISIBLE);
            holder.mDate.setText(DateTimeUtils.parseDateTime(timeLineModel.getDate(), "yyyy-MM-dd HH:mm", "hh:mm a, dd-MMM-yyyy"));
        }
        else
            holder.mDate.setVisibility(View.GONE);

        holder.mMessage.setText(timeLineModel.getMessage());
    }

    @Override
    public int getItemCount() {
        return (mFeedList!=null? mFeedList.size():0);
    }

}

File: app/src/main/java/com/github/vipulasri/timelineview/sample/TimeLineAdapter.kt
package com.github.vipulasri.timelineview.sample

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.github.vipulasri.timelineview.TimelineView
import com.github.vipulasri.timelineview.sample.databinding.ItemTimelineBinding
import com.github.vipulasri.timelineview.sample.databinding.ItemTimelineHorizontalBinding
import com.github.vipulasri.timelineview.sample.model.OrderStatus
import com.github.vipulasri.timelineview.sample.model.Orientation
import com.github.vipulasri.timelineview.sample.model.TimeLineModel
import com.github.vipulasri.timelineview.sample.model.TimelineAttributes
import com.github.vipulasri.timelineview.sample.utils.VectorDrawableUtils
import com.github.vipulasri.timelineview.sample.extentions.formatDateTime
import com.github.vipulasri.timelineview.sample.extentions.setGone
import com.github.vipulasri.timelineview.sample.extentions.setVisible

class TimeLineAdapter(
    private val mFeedList: List<TimeLineModel>,
    private var mAttributes: TimelineAttributes
) : RecyclerView.Adapter<TimeLineAdapter.BaseViewHolder>() {

    private lateinit var mLayoutInflater: LayoutInflater

    override fun getItemViewType(position: Int): Int {
        return TimelineView.getTimeLineViewType(position, itemCount)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        if (!::mLayoutInflater.isInitialized) {
            mLayoutInflater = LayoutInflater.from(parent.context)
        }
        val binding = ItemTimelineBinding.inflate(mLayoutInflater, parent, false)

        return VerticalViewHolder(binding, mAttributes, viewType)

        /*if (mAttributes.orientation == Orientation.HORIZONTAL) {
            val binding = ItemTimelineHorizontalBinding.inflate(mLayoutInflater, parent, false)
            HorizontalViewHolder(binding, mAttributes, viewType)
        } else {
            val binding = ItemTimelineBinding.inflate(mLayoutInflater, parent, false)
            VerticalViewHolder(binding, mAttributes, viewType)
        }*/
    }

    override fun onBindViewHolder(holder: BaseViewHolder, position: Int) {
        val timeLineModel = mFeedList[position]

        when (timeLineModel.status) {
            OrderStatus.INACTIVE -> {
                holder.timeline.marker = VectorDrawableUtils.getDrawable(
                    holder.itemView.context,
                    R.drawable.ic_marker_inactive,
                    mAttributes.markerColor
                )
            }
            OrderStatus.ACTIVE -> {
                holder.timeline.marker = VectorDrawableUtils.getDrawable(
                    holder.itemView.context,
                    R.drawable.ic_marker_active,
                    mAttributes.markerColor
                )
            }
            else -> {
                holder.timeline.setMarker(
                    ContextCompat.getDrawable(holder.itemView.context, R.drawable.ic_marker),
                    mAttributes.markerColor
                )
            }
        }

        if (timeLineModel.date.isNotEmpty()) {
            holder.date.setVisible()
            holder.date.text = timeLineModel.date.formatDateTime("yyyy-MM-dd HH:mm", "hh:mm a, dd-MMM-yyyy")
        } else {
            holder.date.setGone()
        }

        holder.message.text = timeLineModel.message
    }

    override fun getItemCount() = mFeedList.size

    // Base ViewHolder
    abstract class BaseViewHolder(binding: ItemTimelineBinding) : RecyclerView.ViewHolder(binding.root) {
        abstract val date: TextView
        abstract val message: TextView
        abstract val timeline: TimelineView
    }

    // ViewHolder for Vertical Orientation
    class VerticalViewHolder(
        private val binding: ItemTimelineBinding,
        private val mAttributes: TimelineAttributes,
        viewType: Int
    ) : BaseViewHolder(binding) {
        override val date = binding.textTimelineDate
        override val message = binding.textTimelineTitle
        override val timeline = binding.timeline

        init {
            initTimeline(viewType)
        }

        private fun initTimeline(viewType: Int) {
            timeline.initLine(viewType)
            timeline.markerSize = mAttributes.markerSize
            timeline.setMarkerColor(mAttributes.markerColor)
            timeline.isMarkerInCenter = mAttributes.markerInCenter
            timeline.markerPaddingLeft = mAttributes.markerLeftPadding
            timeline.markerPaddingTop = mAttributes.markerTopPadding
            timeline.markerPaddingRight = mAttributes.markerRightPadding
            timeline.markerPaddingBottom = mAttributes.markerBottomPadding
            timeline.linePadding = mAttributes.linePadding
            timeline.lineWidth = mAttributes.lineWidth
            timeline.setStartLineColor(mAttributes.startLineColor, viewType)
            timeline.setEndLineColor(mAttributes.endLineColor, viewType)
            timeline.lineStyle = mAttributes.lineStyle
            timeline.lineStyleDashLength = mAttributes.lineDashWidth
            timeline.lineStyleDashGap = mAttributes.lineDashGap
        }
    }

    // ViewHolder for Horizontal Orientation
    /*class HorizontalViewHolder(
        private val binding: ItemTimelineHorizontalBinding,
        private val mAttributes: TimelineAttributes,
        viewType: Int
    ) : BaseViewHolder(binding) {
        override val date = binding.textTimelineDate
        override val message = binding.textTimelineTitle
        override val timeline = binding.timeline

        init {
            initTimeline(viewType)
        }

        private fun initTimeline(viewType: Int) {
            timeline.initLine(viewType)
            timeline.markerSize = mAttributes.markerSize
            timeline.setMarkerColor(mAttributes.markerColor)
            timeline.isMarkerInCenter = mAttributes.markerInCenter
            timeline.markerPaddingLeft = mAttributes.markerLeftPadding
            timeline.markerPaddingTop = mAttributes.markerTopPadding
            timeline.markerPaddingRight = mAttributes.markerRightPadding
            timeline.markerPaddingBottom = mAttributes.markerBottomPadding
            timeline.linePadding = mAttributes.linePadding
            timeline.lineWidth = mAttributes.lineWidth
            timeline.setStartLineColor(mAttributes.startLineColor, viewType)
            timeline.setEndLineColor(mAttributes.endLineColor, viewType)
            timeline.lineStyle = mAttributes.lineStyle
            timeline.lineStyleDashLength = mAttributes.lineDashWidth
            timeline.lineStyleDashGap = mAttributes.lineDashGap
        }
    }*/
}
